src/merkleDistributor/MerkleDistributor.ts

```diff
@@ -0,0 +1,30 @@
+import { ethers } from "ethers";
+import { DistributionRecipientsWithProofs, DistributionRecipients, MerkleTree } from "./model";
+
+export class MerkleDistributor {
+  static createMerkleDistributionProofs = (recipients: DistributionRecipients, windowIndex: number) => {
+    const recipientLeafs = Object.keys(recipients).map((addr: string) =>
+      MerkleDistributor.createLeaf(addr, recipients[addr].amount, recipients[addr].accountIndex)
+    );
+    const merkleTree = new MerkleTree(recipientLeafs);
+    const recipientsWithProofs = Object.keys(recipients).reduce((acc, recipientAddress, index) => {
+      return {
+        ...acc,
+        [recipientAddress]: {
+          ...recipients[recipientAddress],
+          proofs: merkleTree.getHexProof(recipientLeafs[index]),
+          windowIndex,
+        },
+      };
+    }, {} as DistributionRecipientsWithProofs);
```

Comment:
tiny nit: this might be simpler with just mapping over the entries and then using Object.fromEntries to turn that back into an object. But that may just be my preference because reduce operations always feel more complex than chained maps :)

Reply:
After importing the existing MerkleTree implementation from the contracts repo, the chained maps were not needed anymore. I have a simple `.reduce` call now